src/lib/core/components/visualizations/implementations/MapVisualization.tsx

```diff
@@ -374,63 +363,89 @@ function MapViz(props: VisualizationProps) {
    * Now we deal with the optional second request to pieplot
    */
   const proportionMode = vizConfig.markerType === 'proportion';
-  const pieplotData = usePromise<PieplotData | undefined>(
+  const piePlotResponse = usePromise<PieplotResponse | undefined>(
```

Comment:
Sorry to be pedantic, but could `pieplotResponse` capitalisation be consistent with `pieplotData` below?

Further to my previous comment (before this review) that Dave :+1:ed - don't change the general structure (separating `pieplotResponse`) but use a useMemo and no usePromise.

You'll have to do something with `<MapVEuMap showSpinner ../>` value.  Presumably `pieplotResponse.pending` instead of `pieplotData.pending`.  And obvs check other uses of `.pending` and `.value`.  It's hard to test if it's working at present because the second request comes back just as fast as the first.